src/RequestInsurance/Models/RequestInsurance.php

```diff
@@ -458,6 +468,47 @@ public function getHeadersWithMaskingApplied(): string
         return json_encode($headers, JSON_THROW_ON_ERROR);
     }
 
+    /**
+     * Sets stats field regarding request processing time to the timings field as an array.
+     * @param TransferStats $transferStats
+     *
+     * @return void
+     */
+    public function setTimings(TransferStats $transferStats): void
+    {
+        $handlerStats = $transferStats->getHandlerStats();
+
+        $relevantStats = [
+            'appconnect_time_us'    => $handlerStats['appconnect_time_us']      ?? -1,
+            'connect_time_us'       => $handlerStats['connect_time_us']         ?? -1,
+            'namelookup_time_us'    => $handlerStats['namelookup_time_us']      ?? -1,
+            'pretransfer_time_us'   => $handlerStats['pretransfer_time_us']     ?? -1,
+            'redirect_time_us'      => $handlerStats['redirect_time_us']        ?? -1,
+            'starttransfer_time_us' => $handlerStats['starttransfer_time_us']   ?? -1,
+            'total_time_us'         => $handlerStats['total_time_us']           ?? -1,
+        ];
+
+        $this->timings = json_encode($relevantStats);
+    }
+
+    /**
+     * Gets the total time duration for a request.
+     * Returns -1 if request was not sent / stats not received for some reason.
+     *
+     * @return int
+     */
+    public function getTotalTime()
+    {
+        if (! isset($this->timings)) {
+            return -1;
+        }
+
+        $arrayTimings = json_decode($this->timings, true);
+        $timeInMicroSeconds = $arrayTimings['total_time_us'] ?? -1;
+
+        return $timeInMicroSeconds != -1 ? (int)(floor($timeInMicroSeconds / 1000)) : -1;
```

Comment:
```suggestion
        $arrayTimings = json_decode($this->timings, true);
        $timeInMicroSeconds = $arrayTimings['total_time_us'] ?? null;
        
        if (timeInMicroSeconds === null) {
            return -1;
        }
        
        return (int) floor($timeInMicroSeconds / 1000);
```

Her kan det øge læsbarheden ved at bruge early returns istedet :smile:

Reply:
Det kan jeg godt gå med til :D Synes også den var lidt voldsk